### PR TITLE
Teach hello_services to provide a platform service

### PR DESCRIPTION
**Pull request by abarth**
Currently this service just returns mock data, but it exercises the plumbing.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @tvolkert @jason-simmons

File: examples/hello_services/ios/HelloServices/AppDelegate.m [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#import "LocationProvider.h"
+
+@interface AppDelegate ()
+
+@end
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    FlutterDartProject* project = [[FlutterDartProject alloc] initFromDefaultSourceForConfiguration];
```
Comment (chinmaygarde) [inline examples/hello_services/ios/HelloServices/AppDelegate.m]: The following items are leaked unless we have ARC turned on in this project:

* `FlutterDartProject`
* `UIWindow`
* `FlutterViewController`
* `LocationProvider`

We generally do not use ARC in any of our projects. If its an example, Xcode allows migration a project from ARC to non-ARC.

If we are using ARC, how is `LocationProvider` not being collected at the end of this scope? The memory management seems non-idiomatic.

File: examples/hello_services/ios/HelloServices/LocationProvider.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+
+@interface LocationProvider : NSObject <FlutterMessageListener>
+
+- (void) registerWithController:(FlutterViewController*) controller;
+
+- (NSString*)didReceiveString:(NSString*)message;
```
Comment (chinmaygarde) [inline examples/hello_services/ios/HelloServices/LocationProvider.h]: `- (NSString*)didReceiveString:(NSString*)message;` is part of the `FlutterMessageListener` protocol. You do not need to re-declare it here.

File: examples/hello_services/ios/HelloServices/LocationProvider.m [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "LocationProvider.h"
+
+@interface LocationProvider ()
```
Comment (chinmaygarde) [inline examples/hello_services/ios/HelloServices/LocationProvider.m]: Nit: Empty class extension serves no purpose. You can get rid of this.

**Comment by abarth** [comment]
@chinmaygarde Updated based on our discussion.  PTAL.

**Comment by chinmaygarde** [comment]
lgtm

**Pull request merged**